com.sap.cloud.lm.sl.cf.database/src/test/java/com/sap/cloud/lm/sl/cf/database/migration/executor/DatabaseTableMigrationExecutorTest.java

```diff
@@ -0,0 +1,63 @@
+package com.sap.cloud.lm.sl.cf.database.migration.executor;
+
+import java.sql.SQLException;
+
+import javax.sql.DataSource;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import com.sap.cloud.lm.sl.cf.database.migration.client.DatabaseQueryClient;
+import com.sap.cloud.lm.sl.cf.database.migration.generator.DatabaseTableInsertQueryGenerator;
+import com.sap.cloud.lm.sl.cf.database.migration.metadata.DatabaseTableData;
+
+class DatabaseTableMigrationExecutorTest {
+
+    private static final String TEST_DATABASE_TABLE = "test";
+    private static final DatabaseTableData DATABASE_TABLE_METADATA = null;
+
+    @Mock
+    private DataSource mockSourceDataSource;
+    @Mock
+    private DataSource mockTargetDataSource;
+    @Mock
+    private DatabaseQueryClient mockSourceDatabaseQueryClient;
+    @Mock
+    private DatabaseQueryClient mockTargetDatabaseQueryClient;
+    @Mock
+    private DatabaseTableInsertQueryGenerator mockDatabaseTableInsertQueryGenerator;
+
+    private ImmutableDatabaseTableMigrationExecutor databaseTableMigrationExecutor;
+
+    @BeforeEach
+    public void initialiseDatabaseTableMigrationExecutorAndConfigureMocks() throws SQLException {
+        MockitoAnnotations.initMocks(this);
+        databaseTableMigrationExecutor = ImmutableDatabaseTableMigrationExecutor.builder()
+                                                                                .sourceDataSource(mockSourceDataSource)
+                                                                                .targetDataSource(mockTargetDataSource)
+                                                                                .sourceDatabaseQueryClient(mockSourceDatabaseQueryClient)
+                                                                                .targetDatabaseQueryClient(mockTargetDatabaseQueryClient)
+                                                                                .databaseTableInsertQueryGenerator(mockDatabaseTableInsertQueryGenerator)
+                                                                                .build();
+        Mockito.doReturn(DATABASE_TABLE_METADATA)
+               .when(mockSourceDatabaseQueryClient)
+               .extractTableData(Mockito.anyString());
+        Mockito.doReturn("")
+               .when(mockDatabaseTableInsertQueryGenerator)
+               .generate(DATABASE_TABLE_METADATA);
+    }
+
+    @Test
+    public void testExecuteMigrationInternalWithTestDatabaseTableString() throws SQLException {
+        databaseTableMigrationExecutor.executeMigrationInternal(TEST_DATABASE_TABLE);
```

Comment:
Test `executeMigration` and not `executeMigrationInternal` since the latter is more prone to changes in its interface.